pkg/cache/client.go

```diff
@@ -0,0 +1,204 @@
+	if err != nil {
+		return Options{}, xerrors.Errorf("cache type error: %w", err)
+	}
+
+	var redisOpts RedisOptions
+	if t == TypeRedis {
+		redisTLSOpts, err := NewRedisTLSOptions(redisCACert, redisCert, redisKey)
+		if err != nil {
+			return Options{}, xerrors.Errorf("redis TLS option error: %w", err)
+		}
+		redisOpts = RedisOptions{
+			Backend:    backend,
+			TLS:        redisTLS,
+			TLSOptions: redisTLSOpts,
+		}
+	} else if ttl != 0 {
+		log.Warn("'--cache-ttl' is only available with Redis cache backend")
+	}
+
+	return Options{
+		Type:  t,
+		TTL:   ttl,
+		Redis: redisOpts,
+	}, nil
+}
+
+type RedisOptions struct {
+	Backend    string
+	TLS        bool
+	TLSOptions RedisTLSOptions
+}
+
+// BackendMasked returns the redis connection string masking credentials
+func (o *RedisOptions) BackendMasked() string {
+	endIndex := strings.Index(o.Backend, "@")
+	if endIndex == -1 {
+		return o.Backend
+	}
+
+	startIndex := strings.Index(o.Backend, "//")
+
+	return fmt.Sprintf("%s****%s", o.Backend[:startIndex+2], o.Backend[endIndex:])
+}
+
+// RedisTLSOptions holds the options for redis cache
+type RedisTLSOptions struct {
+	CACert string
+	Cert   string
+	Key    string
+}
+
+func NewRedisTLSOptions(caCert, cert, key string) (RedisTLSOptions, error) {
+	opts := RedisTLSOptions{
+		CACert: caCert,
+		Cert:   cert,
+		Key:    key,
+	}
+
+	// If one of redis option not nil, make sure CA, cert, and key provided
+	if !lo.IsEmpty(opts) {
+		if opts.CACert == "" || opts.Cert == "" || opts.Key == "" {
+			return RedisTLSOptions{}, xerrors.Errorf("you must provide Redis CA, cert and key file path when using TLS")
+		}
+	}
+	return opts, nil
+}
+
+func NewType(backend string) (Type, error) {
+	// "redis://" or "fs" are allowed for now
+	// An empty value is also allowed for testability
+	switch {
+	case strings.HasPrefix(backend, "redis://"):
+		return TypeRedis, nil
+	case backend == "fs", backend == "":
+		return TypeFS, nil
+	default:
+		return "", xerrors.Errorf("unknown cache backend: %s", backend)
+	}
+}
+
+// NewClient returns a new cache client
+func NewClient(opts Options) (*Client, error) {
+	if opts.Type == TypeRedis {
+		log.Info("Redis cache", log.String("url", opts.Redis.BackendMasked()))
+		options, err := redis.ParseURL(opts.Redis.Backend)
+		if err != nil {
+			return nil, err
+		}
+
+		if tlsOpts := opts.Redis.TLSOptions; !lo.IsEmpty(tlsOpts) {
+			caCert, cert, err := GetTLSConfig(tlsOpts.CACert, tlsOpts.Cert, tlsOpts.Key)
+			if err != nil {
+				return nil, err
+			}
+
+			options.TLSConfig = &tls.Config{
+				RootCAs:      caCert,
+				Certificates: []tls.Certificate{cert},
+				MinVersion:   tls.VersionTLS12,
+			}
+		} else if opts.Redis.TLS {
+			options.TLSConfig = &tls.Config{
+				MinVersion: tls.VersionTLS12,
+			}
+		}
+
+		redisCache := NewRedisCache(options, opts.TTL)
+		return &Client{Cache: redisCache}, nil
+	}
+
+	if opts.TTL != 0 {
+		log.Warn("'--cache-ttl' is only available with Redis cache backend")
+	}
```

Comment:
looks like we can leave this warning in one place
https://github.com/aquasecurity/trivy/blob/4ad4b4f79f0ef44e1618c402c7a61166a063b0fd/pkg/cache/client.go#L52-L54

Reply:
Nice catch. Fixed in https://github.com/aquasecurity/trivy/pull/6977/commits/4df7477d724f75adc631806fe6e6dee947e03768.